update_versions.sh

```diff
@@ -1,18 +1,33 @@
-#! /usr/bin/env bash
+#!/bin/sh
 
+set -e
 
-ROOTDIR="$(cd "${0%/*}" 2>/dev/null || exit; pwd)"
-cd $ROOTDIR
-TEMP="$1.$$"
+[ -z "$1" ] && {
+    echo "Usage: $0 <FILE>, <FILE> is unspecified."
+    exit 1
+}
 
-{
-    echo "#include <string>"
-    echo "std::string pegof_version = \"$(git describe --tags --abbrev=8 --dirty)\";"
-    echo "std::string pcc_version = \"$(git -C packcc describe --tags --abbrev=8 --dirty)\";"
-} > "$TEMP"
+cd "${0%/*}"
 
-if [ ! -f "$1" ] || [ "$(md5sum < "$TEMP")" != "$(md5sum < "$1")" ]; then
-    mv "$TEMP" "$1"
-else
-    rm -f "$TEMP"
-fi
+A="$(git           describe --tags --abbrev=8 --dirty || git           rev-parse --short HEAD)"
+B="$(git -C packcc describe --tags --abbrev=8 --dirty || git -C packcc rev-parse --short HEAD)"
```

Comment:
1. Is this needed to make it work with older git version? If so, explicit check would be better then trying (and writing unnecessary error to stderr). 
2. The fallback returns completely different information (only commit hash), which is not that useful.
3. It'd be nicer, if it was a function, taking directory as argument, to avoid duplication. Something like
```
get_version() {
    git -C "$1" describe --tags --abbrev=8 --dirty || git -C "$1" rev-parse --short HEAD
}
```
4. A and B are not readable variable names. It should be something like `PEGOF_VERSION` and `PACKCC_VERSION`